### PR TITLE
Support Foreman 3.9 and newer

### PR DESCRIPTION
**Pull request by laugmanuel**
- migrate to settings DSL
- drop old Foreman version support
- switch to upstream CI
- switch to upstream rubocop

### CODE REVIEW

**Comment by laugmanuel** [comment]
Do you mind having a look @archanaserver ?

File: Rakefile [hunk 1]
```diff
@@ -38,7 +38,7 @@ task default: :test
 begin
   require 'rubocop/rake_task'
   RuboCop::RakeTask.new
-rescue => _
+rescue StandardError => _e
```
Comment (ekohl) [inline Rakefile]:
```suggestion
rescue StandardError
```

File: lib/tasks/foreman_rescue_tasks.rake
Comment (ekohl) [inline lib/tasks/foreman_rescue_tasks.rake:1]: At the end of the file there's code for Jenkins, which can be dropped.

File: lib/tasks/foreman_rescue_tasks.rake [hunk 2]
```diff
@@ -13,7 +15,7 @@ namespace :test do
 end
 
 namespace :foreman_rescue do
-  task :rubocop do
+  task rubocop: :environment do
```
Comment (ekohl) [inline lib/tasks/foreman_rescue_tasks.rake]: This was used with Jenkins, but new CI just uses `Rakefile`

File: locale/gemspec.rb [hunk 3]
```diff
@@ -1,2 +1,4 @@
+# frozen_string_literal: true
+
 # Matches foreman_rescue.gemspec
 _('TODO: Description of ForemanRescue.')
```
Comment (ekohl) [inline locale/gemspec.rb]: This should match the line from the gemspec.

File: test/lib/tasks/seeds_test.rb [hunk 4]
```diff
@@ -27,7 +29,7 @@ class SeedsTest < ActiveSupport::TestCase
     def seed
       User.current = FactoryBot.build(:user, :admin => true,
                                              :organizations => [], :locations => [])
-      load Rails.root.join('db', 'seeds.rb')
+      load Rails.root.join('db/seeds.rb')
```
Comment (ekohl) [inline test/lib/tasks/seeds_test.rb:32]: Why this change? Why is it better?

File: app/models/setting/rescue.rb [hunk 5]
```diff
@@ -1,45 +1,13 @@
+# frozen_string_literal: true
+
 class Setting
   class Rescue < ::Setting
```
Comment (ekohl) [inline app/models/setting/rescue.rb]: Should this now be a service or some other helper?

File: app/models/setting/rescue.rb [hunk 6]
```diff
@@ -1,45 +1,13 @@
...
-
-      true
-    end
-
     def self.templates(kind)
       template_kind = TemplateKind.find_by(name: kind)
       templates = ProvisioningTemplate.where(:template_kind => template_kind)
       templates.each_with_object({}) do |template, hsh|
         hsh[template.name] = template.name
       end
```
Comment (ekohl) [inline app/models/setting/rescue.rb]: Would this work?
```suggestion
      ProvisioningTemplate.where(:template_kind => template_kind).pluck(:name, :name).to_h
```

File: lib/foreman_rescue/engine.rb [hunk 7]
```diff
@@ -51,7 +65,7 @@ class Engine < ::Rails::Engine
     end
 
     initializer 'foreman_rescue.register_gettext', after: :load_config_initializers do |_app|
-      locale_dir = File.join(File.expand_path('../../..', __FILE__), 'locale')
+      locale_dir = File.join(File.expand_path('../..', __dir__), 'locale')
```
Comment (ekohl) [inline lib/foreman_rescue/engine.rb:69]: Isn't this essentially the same?
```suggestion
      locale_dir = File.expand_path('../../locale', __dir__)
```

File: test/lib/tasks/seeds_test.rb (hunk 4 above)
Comment (laugmanuel) [reply test/lib/tasks/seeds_test.rb:32]: `EnforcedStyle: slashes` is the default for https://www.rubydoc.info/gems/rubocop-rails/2.5.2/RuboCop/Cop/Rails/FilePath

File: lib/foreman_rescue/engine.rb (hunk 7 above)
Comment (laugmanuel) [reply lib/foreman_rescue/engine.rb:69]: Probably is. this was autocorrected by rubocop due to https://www.rubydoc.info/gems/rubocop/RuboCop/Cop/Style/ExpandPathArguments

File: test/lib/tasks/seeds_test.rb (hunk 4 above)
Comment (ekohl) [reply test/lib/tasks/seeds_test.rb:32]: That is surprising

**Review by ekohl — APPROVED** [review]

**Comment by laugmanuel** [comment]
Thanks for the review @ekohl !
